test-lib.sh

```diff
@@ -456,13 +456,35 @@ ct_s2i_build_as_df()
     local user_id=
     local df_name=
     local tmpdir=
+    local incremental=false
     # Use /tmp to not pollute cwd
     tmpdir=$(mktemp -d)
     df_name=$(mktemp -p "$tmpdir" Dockerfile.XXXX)
     pushd "$tmpdir"
     # Check if the image is available locally and try to pull it if it is not
     docker images "$src_image" &>/dev/null || echo "$s2i_args" | grep -q "pull-policy=never" || docker pull "$src_image"
-    user_id=$(docker inspect -f "{{.Config.User}}" "$src_image")
+    user=$(docker inspect -f "{{.Config.User}}" "$src_image")
+    # Default to root if no user is set by the image
+    user=${user:-0}
+    # run the user through the image in case it is non-numeric or does not exist
+    user_id=$(docker run --rm "$src_image" bash -c "id -u $user" 2>/dev/null || echo "notfound")
```

Comment:
Except that this happens always, even if that is numeric already (or default == 0).  It is probably OK to do `id -u 26` (gives 26), but we can save one container run if we've "only run through the image  in case it is non-numeric".

Reply:
nit, slightly nicer is `if ! user_id=$(docker run ... 2>/dev/null); then`, you can avoid the "echo notfound" which might be theoretically a real user :-).

Reply:
id -u returns number.

Reply:
>nit, slightly nicer is if ! user_id=$(docker run ... 2>/dev/null); then, you can avoid the "echo notfound" which might be theoretically a real user :-).

Would `id -u ...` be able to return `notfound`? But I agree that it looks cleaner so will fix.

>Except that this happens always, even if that is numeric already (or default == 0).

I did not add the check as I thought it might be able to pick up a user that is numeric and not located in the image's `/etc/passwd` but now that I think I am not sure if that can even happen, so will add the check. Thanks!

Reply:
right, `id -u ..` should only return numbers, ... so this is nothing more than a "style"-comment, sorry....